picard.go

```diff
@@ -457,7 +488,11 @@ func (p PersistenceORM) checkForExisting(
 				}
 			}
 		}
-		query = query.Where(strings.Join(wheres, " || '"+separator+"' || ")+" = ANY(?)", pq.Array(lookupObjectKeys))
+		if nonExisting {
+			query = query.Where("NOT "+strings.Join(wheres, " || '"+separator+"' || ")+" = ANY(?)", pq.Array(lookupObjectKeys))
```

Comment:
I'm not a giant fan of
a) this style of string concatenation instead of `fmt.Sprintf()` or
b) having this instead of just `if nonExisting { queryStr = fmt.Sprintf("NOT %v", queryStr) }` and then `query = query.Where(queryStr, pq.Array(lookupObjectKeys))`.
But since this has been out here for months and I don't think Fabian is going to change it, let's just leave it for now.